File: Language_Essentials/python/q1_list_operations.py
# List/Array Operations - TCS NQT

def process_list():
    # 1. Initialization
    lst = [10, 20, 30]
    
    # 2. Add elements
    lst.append(40)
    lst.append(50)
    
    print("Initial list:", lst)
    
    # 3. Access elements
    print("First element:", lst[0])
    print("Last element:", lst[-1])
    print("Element at index 2:", lst[2])
    
    # 4. Remove elements
    lst.pop() # Removes 50
    print("After pop:", lst)
    
    # 5. Size 
    print("Size:", len(lst))
    
    # 6. Sorting
    lst2 = [5, 2, 8, 1, 9]
    lst2.sort()
    print("Sorted lst2:", lst2)
    
    # 7. Reversing
    lst2.reverse() # In-place
    print("Reversed lst2:", lst2)

if __name__ == "__main__":
    process_list()

File: Language_Essentials/java/Q1ListOperations.java
// ArrayList Operations - TCS NQT
import java.util.ArrayList;
import java.util.Collections;

public class Q1ListOperations {
    public static void printList(ArrayList<Integer> list) {
        for (int num : list) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        // 1. Initialization
        ArrayList<Integer> list = new ArrayList<>();
        list.add(10);
        list.add(20);
        list.add(30);
        
        // 2. Add elements
        list.add(40);
        list.add(50);
        
        System.out.print("Initial list: ");
        printList(list);
        
        // 3. Access elements
        System.out.println("First element: " + list.get(0));
        System.out.println("Last element: " + list.get(list.size() - 1));
        System.out.println("Element at index 2: " + list.get(2));
        
        // 4. Remove elements
        list.remove(list.size() - 1); // Removes 50 (index)
        System.out.print("After remove: ");
        printList(list);
        
        // 5. Size
        System.out.println("Size: " + list.size());
        
        // 6. Sorting
        ArrayList<Integer> list2 = new ArrayList<>();
        Collections.addAll(list2, 5, 2, 8, 1, 9);
        Collections.sort(list2);
        System.out.print("Sorted list2: ");
        printList(list2);
        
        // 7. Reversing
        Collections.reverse(list2);
        System.out.print("Reversed list2: ");
        printList(list2);
    }
}
